ts/sagas/api.ts

```diff
@@ -0,0 +1,37 @@
+import { Either, left, right } from "fp-ts/lib/Either";
+import { Effect } from "redux-saga";
+import { call, put } from "redux-saga/effects";
+
+import { BasicResponseTypeWith401 } from "../api/backend";
+import { sessionExpired } from "../store/actions/authentication";
+
+/**
+ * A generator that wrap an API Call and handle the response status.
+ * If the response status is 401 the user must be logged out.
+ */
+export function* callApi<
```

Comment:
this assumes a call to a certain API that may return 401, I would name this method in a way that is more specific to the use case